diesel/src/types/mod.rs

```diff
@@ -337,18 +338,100 @@ pub enum IsNull {
     No,
 }
 
+#[derive(Debug, Clone, Copy)]
+#[doc(hidden)]
+pub struct ToSqlOutput<'a, T, DB> where
```

Comment:
In my opinion this struct wraps two totally unrelated things. The only common point here is that both parts are passed to `ToSql::to_sql`. I don't see any advantage over passing those fields directly as arguments to `to_sql`. 
Creating this wrapping struct needs ~100 lines boilerplate code for nothing (as far as I understand).